dockstore-launcher/src/main/java/io/dockstore/common/FileProvisionUtil.java

```diff
@@ -138,35 +148,114 @@ public static String getFilePluginLocation(INIConfiguration config) {
-        Path pluginPath = Paths.get(filePluginLocation, String.format("%2$s-%1$s.zip", version,
-                pluginName));
-        String icgcLocation = icgcUrl.toString();
-        String pluginLocation = pluginPath.toString();
+        String pluginZip = String.format("%2$s-%1$s.zip", version, pluginName);
+        Path pluginPath = Paths.get(filePluginLocation, pluginZip);
+        String destinationLocation = pluginPath.toString();
         if (Files.exists(pluginPath)) {
-            System.out.println("Skipping " + pluginLocation + ", already exists");
+            System.out.println("Skipping " + destinationLocation + ", already exists");
+            return false;
         } else {
-            System.out.println("Downloading " + icgcLocation + " to " + pluginLocation);
-            FileProvisionUtil.downloadFromVFS2(icgcLocation, pluginLocation);
+            System.out.println("Downloading " + sourceLocation + " to " + destinationLocation);
+            FileProvisionUtil.downloadFromVFS2(sourceLocation, destinationLocation);
+            return true;
+        }
+    }
+
+    /**
+     * Extracts plugin information from json and then downloads the plugin
+     *
+     * @param filePluginLocation The location of the plugins folder
+     * @param json               The PluginJSON object
+     */
+    private static boolean downloadPlugin(String filePluginLocation, PluginJSON json) {
+        try {
+            LOG.info("Downloading Plugins");
+            String version = json.getVersion();
+            String name = json.getName();
+            String sourceLocation;
+            // A location parameter in the json object indicates it's not on oicr artifactory
+            if (json.getLocation() == null) {
+                String template = "https://artifacts.oicr.on.ca/artifactory/collab-release/io/dockstore/%2$s/%1$s/%2$s-%1$s.zip";
+                URL sourceURL = new URI(String.format(template, version, name)).toURL();
+                sourceLocation = sourceURL.toString();
+            } else {
+                sourceLocation = json.getLocation();
+            }
+            downloadPlugin(filePluginLocation, version, name, sourceLocation);
+            return true;
+        } catch (MalformedURLException | URISyntaxException e) {
+            LOG.error("Could not download plugin: " + e.getMessage());
+            return false;
+        }
+    }
+
+    /**
+     * Creates the plugins json file
+     *
+     * @param location Location of where to create the file
+     */
+    static boolean createPluginJSONFile(String location) {
+        InputStream in = FileProvisionUtil.class.getResourceAsStream("/" + PLUGINS_JSON_FILENAME);
+        File targetFile = new File(location);
+        try {
+            FileUtils.copyInputStreamToFile(in, targetFile);
+            return true;
+        } catch (IOException e) {
+            LOG.error("Could not create " + PLUGINS_JSON_FILENAME);
+            e.printStackTrace();
```

Comment:
Pass "e" into the LOG call and remove the explicit call to print the stack trace. 
(Avoid calling  e.printStackTrace();)